hearth/lib/hearth/middleware/send.rb

```diff
@@ -71,25 +71,69 @@ def call(input, context)
       private
 
       def stub_response(input, context, output)
-        stub = @stubs.next(context.operation_name)
-        log_debug(context, "Stubbing response with stub: #{stub}")
-        apply_stub(stub, input, context, output)
-        log_debug(context, "Stubbed response: #{context.response.inspect}")
-        return unless context.response.body.respond_to?(:rewind)
-
-        context.response.body.rewind
+        context.tracer.in_span(
+          'Middleware.Stub',
+          attributes: request_attributes(context)
+        ) do |span|
+          stub = @stubs.next(context.operation_name)
+          log_debug(context, "Stubbing response with stub: #{stub}")
+          apply_stub(stub, input, context, output)
+          log_debug(context, "Stubbed response: #{context.response.inspect}")
+          response_attributes(span, context)
+        end
+        rewind_response_body(context)
       end
 
       def send_request(context, output)
-        log_debug(context, "Sending request: #{context.request.inspect}")
-        @client.transmit(
-          request: context.request,
-          response: context.response,
-          logger: context.config.logger
-        )
-        log_debug(context, "Received response: #{context.response.inspect}")
-      rescue Hearth::NetworkingError => e
-        output.error = e
+        context.tracer.in_span(
+          'Middleware.Send',
+          attributes: request_attributes(context)
+        ) do |span|
+          log_debug(context, "Sending request: #{context.request.inspect}")
+          @client.transmit(
+            request: context.request,
+            response: context.response,
+            logger: context.config.logger
+          )
+          response_attributes(span, context)
+          log_debug(context, "Received response: #{context.response.inspect}")
+        rescue Hearth::NetworkingError => e
+          output.error = e
+        end
+      end
+
+      def request_attributes(context)
+        attributes =
+          {
+            'http.method' => context.request.http_method,
+            'net.protocol.name' => 'http',
+            'net.protocol.version' => Net::HTTP::HTTPVersion,
+            'net.peer.name' => context.request.uri.host,
+            'net.peer.port' => context.request.uri.port
+          }
+        if context.request.headers.key?('Content-Length')
+          attributes['http.request_content_length'] =
+            context.request.headers['Content-Length']
+        end
+        attributes
+      end
+
+      def response_attributes(span, context)
+        attributes =
+          {
+            'http.status_code' => context.response.status
+          }
+        if context.response.headers.key?('Content-Length')
+          attributes['http.response_content_length'] =
+            context.response.headers['Content-Length']
```

Comment:
Assumption here is that the response will contain this header that I could use to set this span attribute. I'm not sure if I should take this a step further to check the `body.size` (if possible) when there is none set.